octobot_commons/databases/cache_manager.py

```diff
@@ -0,0 +1,178 @@
+#  Drakkar-Software OctoBot-Commons
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import os
+
+import octobot_commons.logging as logging
+import octobot_commons.databases.document_database_adaptors as adaptors
+import octobot_commons.databases.implementations.cache_timestamp_database as cache_timestamp_database
+import octobot_commons.databases.databases_util as databases_util
+import octobot_commons.constants as common_constants
+import octobot_commons.symbol_util as symbol_util
+import octobot_commons.errors as common_errors
+import octobot_commons.event_tree as event_tree
+
+
+class CacheManager:
+    """
+    Manages cache as a global dict since caches can be accessed from live, backtesting and optimizers concurrently
+    """
+
+    CACHES = event_tree.EventTree()
+    DEFAULT_CONFIG_IDENTIFIER = "default"
+
+    def __init__(self, database_adaptor=adaptors.TinyDBAdaptor):
+        self.database_adaptor = database_adaptor
+        self.logger = logging.get_logger(self.__class__.__name__)
+
+    def get_cache(self, tentacle, tentacle_name, exchange_name, symbol, time_frame, config_name,
```

Comment:
Does symbol or timeframe can be null? In other word, should we put default values with empty strings for timeframe, symbol etc?